### PR TITLE
Issue #207: Add `email` back into `add` command

### LINKED ISSUE #207: Refactor `add` command to allow user to input `email`

### PR DESCRIPTION
**Pull request by domlimm**
Closes #207.

### CODE REVIEW

File: src/main/java/seedu/address/model/candidate/Email.java [hunk 1]
```diff
@@ -29,8 +19,9 @@
             + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
     private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
     private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
-    public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@" + DOMAIN_REGEX;
+    private static final String USER_NAME = "[E][\\p{Digit}]{7}";
```
Comment (lzan98) [inline src/main/java/seedu/address/model/candidate/Email.java:22]: 👍

**Review by lzan98 — APPROVED** [review]
LGTM! Side note, may I ask why we are adding back email? Is it because of problems with editing studentId and email not updating?

**Comment by domlimm** [comment]
@lzan98 yes that's correct leon! Thanks for reviewing!

@charmainehly and I had a talk on how we can overcome it. We're changing `studentId` to `AXXXXXXXX` and we'll restrict email to be `EXXXXXXX@u.nus.edu`. To differentiate between 2 candidates - `studentId` and `email` have to be unique.

So now, all fields are editable without any underlying issues i.e. Re-using what AB3 has, just to 'workaround' the codebase :')

**Review by charmainehly — APPROVED** [review]
lgtm thanks dom!!

**Pull request merged**